sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/plugin/InputFileContextImplTest.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.plugin
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import org.sonar.api.batch.fs.InputFile
+import org.sonar.api.batch.sensor.SensorContext
+import org.sonar.api.batch.sensor.issue.MessageFormatting
+import org.sonar.api.batch.sensor.issue.NewIssue
+import org.sonar.api.batch.sensor.issue.NewIssueLocation
+import org.sonar.api.batch.sensor.issue.NewMessageFormatting
+import org.sonar.api.rule.RuleKey
+import org.sonarsource.kotlin.api.message
+
+class InputFileContextImplTest {
+    @Test
+    fun `ensure message formatting is included in issue report`() {
+        val newMessageFormatting = spyk<NewMessageFormatting>()
+        val newLocation = mockk<NewIssueLocation> {
+            every { newMessageFormatting() } returns newMessageFormatting
+            every { on(any()) } returns this
+            every { message(any()) } returns this
+            every { message(any(), any()) } returns this
+        }
+        val newIssue = mockk<NewIssue> {
+            every { newLocation() } returns newLocation
+            every { forRule(any()) } returns this
+            every { at(any()) } returns this
+            every { gap(any()) } returns this
+            every { save() } returns Unit
+        }
+
+        val sensorContext = mockk<SensorContext> {
+            every { newIssue() } returns newIssue
+        }
+
+        val inputFileContextImpl = InputFileContextImpl(sensorContext, mockk<InputFile>(), false)
```

Comment:
I think it would be more readable if this would be a field and you make all the setup in a separate setup function. Wdyt?

Reply:
I am not sure what you are envisioning. The difficulty here is that I need to call something on the `inputFileContextImpl` to then assert something on `newLocation` and `newMessageFormatting`. So I need to use quite a few of these vals after setup - I am not convinced taking them out of the function would help readability :thinking:

Reply:
Okay, maybe that could help to make it more readable